yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -163,6 +163,23 @@ private[spark] class Client(
     appContext.setQueue(args.amQueue)
     appContext.setAMContainerSpec(containerContext)
     appContext.setApplicationType("SPARK")
+    sparkConf.getOption(CONF_SPARK_YARN_APPLICATION_TAGS)
```

Comment:
Not a big deal, but I'd flip this around and do something like:

    Try(appContext.getClass().getMethod("setApplicationTags", classOf[java.util.Set[String]])).foreach { setTagsMethod =>
      ...
    }

That way you print the log message only once, lookup the method only once, and skip all the processing of the option if the method is not available.